core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java

```diff
@@ -333,7 +332,20 @@ public static RegisteredType addAliases(RegisteredType type, Iterable<String> al
     @Beta
     public static RegisteredType addTag(RegisteredType type, Object tag) {
         if (tag!=null) {
-            ((BasicRegisteredType)type).tags.add( tag );
+            if (tag instanceof Map &&( (Map<String,Object>) tag).containsKey(BrooklynTags.YAML_SPEC_HIERARCHY)) {
```

Comment:
special-casing this tag here feels wrong.  feels like the logic belongs in `BrooklynTags` or `HierarchySpecTag`.